daemon/internal/cloud_client/client_state.go

```diff
@@ -0,0 +1,76 @@
+package cloud_client
+
+import (
+	"github.com/akitasoftware/akita-cli/rest"
+	"github.com/akitasoftware/akita-libs/akid"
+	"github.com/akitasoftware/akita-libs/daemon"
+)
+
+// Logging state for a single service.
```

Comment:
It would have been helpful, for my understanding, to have a comment here along the lines "all the modifications to this are done by a single worker goroutine" because I started thinking about race conditions.

I was worried about the large number of events making modifications to serviceInfo state, and was going to suggest they be moved into their own functions for encapsulation purposes, but I think I understand the structure better now.

Reply:
Done in 1190f53.